components/filter/filter.js

```diff
@@ -0,0 +1,270 @@
+					<d2l-button-icon
+						@click="${this._handleDimensionHide}"
+						icon="tier1:chevron-left"
+						text="${this.localize('components.menu-item-return.returnCurrentlyShowing', 'menuTitle', dimensionText)}">
+					</d2l-button-icon>
+					<div class="d2l-filter-dimension-header-text d2l-body-standard">${dimensionText}</div>
+				</div>
+			</div>
+		`;
+	}
+
+	_createSetDimension(dimension) {
+		return html`
+			<d2l-list
+				@d2l-list-selection-change="${this._handleChangeSetDimension}"
+				extend-separators>
+
+				${dimension.values.map(item => html`
+					<d2l-list-item
+						key="${item.key}"
+						selectable
+						?selected="${item.selected}"
+						slim>
+
+						<div class="d2l-filter-dimension-set-value-text">${item.text}</div>
+					</d2l-list-item>
+				`)}
+			</d2l-list>
+		`;
+	}
+
+	_dispatchChangeEvent(eventDetail) {
+		this.dispatchEvent(new CustomEvent('d2l-filter-change', { bubbles: false, composed: false, detail: eventDetail }));
+	}
+
+	_getSlottedNodes(slot) {
+		const nodes = slot.assignedNodes();
+		return nodes.filter((node) => node.nodeType === Node.ELEMENT_NODE && node.tagName.toLowerCase() !== 'template');
+	}
+
+	_handleChangeSetDimension(e) {
+		const dimensionKey = e.composedPath()[0].parentNode.id;
+		const valueKey = e.detail.key;
+		const selected = e.detail.selected;
+
+		// Update the corresponding d2l-filter-dimension-set-value to keep them in sync
+		const dimension = this.querySelector(`d2l-filter-dimension-set[key="${dimensionKey}"`);
+		const slot = dimension.shadowRoot.querySelector('slot');
+		const items = this._getSlottedNodes(slot);
+		const item = items.find(item => item.key === valueKey);
+		item.selected = selected;
+
+		this._dispatchChangeEvent({ dimension: dimensionKey, value: { key: valueKey, selected: selected } });
+	}
+
+	_handleDimensionDataChange(e) {
+		const dimension = this._dimensions.find(dimension => dimension.key === e.detail.dimensionKey);
+		const value = e.detail.valueKey && dimension.values.find(value => value.key === e.detail.valueKey);
+		const toUpdate = value ? value : dimension;
+		const changes = e.detail.changes;
+
+		changes.forEach((newValue, prop) => {
+			if (toUpdate[prop] !== newValue) {
+				toUpdate[prop] = newValue;
+				this.requestUpdate();
+			}
+		});
+	}
+
+	_handleDimensionHide() {
+		this.shadowRoot.querySelector(`#${this._activeDimensionKey}`).hide();
+		this._activeDimensionKey = null;
+	}
+
+	_handleDimensionShowComplete() {
+		const dimension = this.shadowRoot.querySelector(`#${this._activeDimensionKey}`);
+		const focusable = getFirstFocusableDescendant(dimension);
+		if (focusable) {
+			focusable.focus();
+		}
+	}
+
+	_handleDimensionShowStart(e) {
+		this._activeDimensionKey = e.detail.sourceView.id;
+	}
+
+	_handleDropdownClose(e) {
+		this._activeDimensionKey = null;
+		this._stopPropagation(e);
+	}
+
+	_handleSlotChange() {
+		const dimensionNodes = this._getSlottedNodes(this._dimensionsSlot);
```

Comment:
This can just be:
```javascript
const dimensionNodes = this._getSlottedNodes(e.target);
```
... which means the `_dimensionsSlot` variable currently isn't needed.